src/Core/src/Platform/iOS/KeyboardAutoManagerScroll.cs

```diff
@@ -0,0 +1,614 @@
+							move = (nfloat)Math.Min(0, previousCellRectInRootSuperview.GetMaxY() - topLayoutGuide);
+						}
+					}
+				}
+
+				else if (superScrollView.FindResponder<UICollectionView>() is UICollectionView collectionView)
+				{
+					shouldContinue = superScrollView.ContentOffset.Y > 0;
+
+					if (shouldContinue && View.FindResponder<UICollectionViewCell>() is UICollectionViewCell collectionCell
+						&& collectionView.IndexPathForCell(collectionCell) is NSIndexPath indexPath
+						&& collectionView.GetPreviousIndexPath(indexPath) is NSIndexPath previousIndexPath
+						&& collectionView.GetLayoutAttributesForItem(previousIndexPath) is UICollectionViewLayoutAttributes attributes)
+					{
+						var previousCellRect = attributes.Frame;
+
+						if (!previousCellRect.IsEmpty)
+						{
+							var previousCellRectInRootSuperview = collectionView.ConvertRectToView(previousCellRect, RootController.Superview);
+							move = (nfloat)Math.Min(0, previousCellRectInRootSuperview.GetMaxY() - topLayoutGuide);
+						}
+					}
+				}
+
+				else
+				{
+					if (cursorRect.Y - innerScrollValue >= topLayoutGuide && cursorRect.Y - innerScrollValue <= keyboardYPosition)
+						shouldContinue = false;
+					else
+						shouldContinue = true;
+
+					if (cursorRect.Y - innerScrollValue < topLayoutGuide)
+						move = cursorRect.Y - innerScrollValue - (nfloat)topLayoutGuide;
+					else if (cursorRect.Y - innerScrollValue > keyboardYPosition)
+						move = cursorRect.Y - innerScrollValue - keyboardYPosition;
+				}
+
+				// Go up the hierarchy and look for other scrollViews until we reach the UIWindow
+				if (shouldContinue)
+				{
+					var tempScrollView = superScrollView.FindResponder<UIScrollView>();
+					UIScrollView? nextScrollView = null;
+
+					// set tempScrollView to next scrollable superview of superScrollView
+					while (tempScrollView is not null)
+					{
+						if (tempScrollView.ScrollEnabled)
+						{
+							nextScrollView = tempScrollView;
+							break;
+						}
+						tempScrollView = tempScrollView.FindResponder<UIScrollView>();
+					}
+
+					var shouldOffsetY = superScrollView.ContentOffset.Y - Math.Min(superScrollView.ContentOffset.Y, -move);
+
+					// the contentOffset.Y will change to shouldOffSetY so we can subtract the difference from the move
+					move -= (nfloat)(shouldOffsetY - superScrollView.ContentOffset.Y);
+
+					var newContentOffset = new CGPoint(superScrollView.ContentOffset.X, shouldOffsetY);
+
+					if (!superScrollView.ContentOffset.Equals(newContentOffset))
+					{
+						if (nextScrollView is null)
+						{
+							AnimateScroll(() =>
+							{
+								newContentOffset.Y += innerScrollValue;
+								innerScrollValue = 0;
+
+								if (View.FindResponder<UIStackView>() is not null)
+									superScrollView.SetContentOffset(newContentOffset, UIView.AnimationsEnabled);
+								else
+									superScrollView.ContentOffset = newContentOffset;
```

Comment:
This is a closure that is capturing the superScrollView which is I suppose an UIView, therefore is a memory leak.

Reply:
Mea culpa, should have said increased memory pressure. We are creating an extra object with a reference to the capture elements. The gc will have extra work.